File: exp-04/q3-naive-string-matching/solution.py
text = input()
pattern = input()
n, m = len(text), len(pattern)

found = False
for i in range(n - m + 1):
    j = 0
    while j < m and text[i + j] == pattern[j]:
        j += 1
    if j == m:
        print(f"Pattern found at index {i}")
        found = True

if not found:
    print("not found")

File: exp-04/q3-naive-string-matching/solution.js
const readline = require('readline');

const rl = readline.createInterface({ input: process.stdin, output: process.stdout });

let lines = [];
rl.on('line', (line) => {
    lines.push(line);
    if (lines.length === 2) {
        const text = lines[0];
        const pattern = lines[1];
        const n = text.length, m = pattern.length;
        let found = false;
        for (let i = 0; i <= n - m; i++) {
            let j = 0;
            while (j < m && text[i + j] === pattern[j]) j++;
            if (j === m) {
                console.log(`Pattern found at index ${i}`);
                found = true;
            }
        }
        if (!found) console.log('not found');
        rl.close();
    }
});